eng/common/scripts/Package-Properties.ps1

```diff
@@ -164,21 +159,45 @@ function Get-CSVMetadata ([string]$MetadataUri=$MetadataUri)
     return $metadataResponse
 }
 
-function Operate-OnPackages ($activePkgList, $ServiceDirectory, [Array]$pkgPropsResult)
+function Get-PkgPropsForEntireService ($serviceDirectoryPath, $serviceDirectoryName)
 {
-    foreach ($pkg in $activePkgList)
+    $projectProps = @() # Properties from very project inthe service
+    $packageProps = @() # Properties for artifacts specified in ci.yml
+
+    foreach ($directory in (Get-ChildItem $serviceDirectoryPath -Directory))
+    {
+        $pkgDirectoryPath = Join-Path $serviceDirectoryPath $directory.Name
+
+        if ($GetPackageInfoFromRepoFn -and (Test-Path "Function:$GetPackageInfoFromRepoFn"))
+        {
+            $pkgProps = &$GetPackageInfoFromRepoFn -pkgDirectoryPath $pkgDirectoryPath -serviceDirectory $serviceDirectoryName
```

Comment:
`$pkgProps = &$GetPackageInfoFromRepoFn -pkgPath $pkgDirectoryPath -serviceDirectory $ServiceDirectory -pkgName $PackageName`

That was the older signature so I think we need to have to figure out how to still work with that older signature until we can get the language repos to support the new signature. I suggest we start by keeping the first 2 parameters the same and then we need to figure out how we make the languages work correct if null is passed for the last parameter (ie. we don't pass it). We probably need to go and fix all the language repo's to somehow work with pkgName being null before we update this. Or we have to create a new function that we can switch to temporarily and then remove the old function.

Reply:
I plan to update the language end then stage the merge so that its just a few seconds apart. The change here is tooo significant that we have to update the logic in the language repos because we are no longer using the artifact names directly.